canonical-web-and-design/maas-ui
# feat(ui): add hugepage + interface data to KVM aggregate view if KVM is NUMA-aware

## Done

- If a KVM has `numa_pinning` data, it will be summed and added to the KVM aggregate view. If not, only the data on the base KVM object will be used.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the KVM details page and check that the aggregate data for hugepages, virtual functions and interface names matches the fake NUMA api data. Note that memory and cores will **not** match the fake NUMA data, because it uses real data.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2083
